src/NuGet.Clients/NuGet.CommandLine/Common/SelfUpdater.cs

```diff
@@ -121,7 +121,7 @@ internal async Task UpdateSelfFromVersionAsync(string exePath, bool prerelease,
 
                         using (Stream fromStream = await downloadResourceResult.PackageReader.GetStreamAsync(nugetExeInPackageFilePath, cancellationToken), toStream = File.Create(exePath))
                         {
-                            fromStream.CopyTo(toStream);
+                            await fromStream.CopyToAsync(toStream);
```

Comment:
The project `NuGet.CommandLine` is what gets compiled into `nuget.exe`, which only runs on the .NET Framework, not .NET (Core). From all my testing of async file access on .NET Framework, async file IO is "considerably" (relatively speaking, from benchmarks) slower than sync file IO. Granted, this code path will only copy a single file, so it won't be measurably different to humans. But the point is that async file IO is not universally better.